starlite/openapi/controller.py

```diff
@@ -65,6 +66,31 @@ def get_schema_from_request(request: Request) -> "OpenAPI":
             raise ImproperlyConfiguredException("Starlite has not been instantiated with OpenAPIConfig")
         return request.app.openapi_schema
 
+    @staticmethod
+    def should_serve_endpoint(request: Request) -> bool:
+        """This method verifies that the requested path is within the allowed
+        endpoints in the openapi_config.
+
+        Returns:
+            A boolean.
+
+        Raises:
+            [ImproperlyConfiguredException][starlite.exceptions.ImproperlyConfiguredException]: If the application
+                `openapi_config` attribute is `None`.
+        """
+        if not request.app.openapi_config:  # pragma: no cover
+            raise ImproperlyConfiguredException("Starlite has not been instantiated with OpenAPIConfig")
+
+        path = set(filter(None, request.url.path.split("/")))
+
+        config = request.app.openapi_config
+        if path == {"schema"} and config.root_schema_site in config.allowed_endpoints:
```

Comment:
This is not going to work. A user can subclass the controller and change the root path, e.g. from `/schema` to `/api-docs`. Additionally the user can override the methods to change their paths.

Reply:
You're right, I will rethink how to validate when accessing the root, in case root_schema_site is not in enabled_endpoints (which should not happen, but I still think it is important to check it).